components/prism-arturo.js

```diff
@@ -0,0 +1,50 @@
+Prism.languages.arturo = {
+    'comment': /;.*/,
+
+    'character': {
+        pattern: /`.`/,
+        alias: 'function'
+    },
+
+    'string': {
+        pattern: /"(?:[^"\\]|\\.)*"/,
+        greedy: true,
+        alias: 'function'
+    },
+
+    'label': {
+        pattern: /[\w]+\b\??:/i,
+        alias: 'builtin'
+    },
+
+    'literal': {
+        pattern: /'(?:[\w]+\b\??:?)/,
+        alias: 'boolean'
+    },
+
+    'type': {
+        pattern: /:(?:[\w]+\b\??:?)/,
+        alias: 'boolean'
+    },
+
+    'keyword1': {
+        pattern: /\b(?:all|and|any|ascii|attr|attribute|attributeLabel|binary|block|char|contains|database|date|dictionary|empty|equal|even|every|exists|false|floating|function|greater|greaterOrEqual|if|in|inline|integer|is|key|label|leap|less|lessOrEqual|literal|logical|lower|nand|negative|nor|not|notEqual|null|numeric|odd|or|path|pathLabel|positive|prefix|prime|regex|same|set|some|sorted|standalone|string|subset|suffix|superset|symbol|symbolLiteral|true|try|type|unless|upper|when|whitespace|word|xnor|xor|zero)\?/,
+        alias: 'keyword'
+    },
+
+    'keyword2': {
+        pattern: /\b(?:abs|acos|acosh|acsec|acsech|actan|actanh|add|after|alert|alias|and|angle|append|arg|args|arity|array|as|asec|asech|asin|asinh|atan|atan2|atanh|attr|attrs|average|before|benchmark|blend|break|builtins1|builtins2|call|capitalize|case|ceil|chop|clear|clip|close|color|combine|conj|continue|copy|cos|cosh|crc|csec|csech|ctan|ctanh|cursor|darken|dec|decode|define|delete|desaturate|deviation|dialog|dictionary|difference|digest|digits|div|do|download|drop|dup|e|else|empty|encode|ensure|env|escape|execute|exit|exp|extend|extract|factors|false|fdiv|filter|first|flatten|floor|fold|from|function|gamma|gcd|get|goto|hash|hypot|if|inc|indent|index|infinity|info|input|insert|inspect|intersection|invert|jaro|join|keys|kurtosis|last|let|levenshtein|lighten|list|ln|log|loop|lower|mail|map|match|max|maybe|median|min|mod|module|mul|nand|neg|new|nor|normalize|not|now|null|open|or|outdent|pad|palette|panic|path|pause|permissions|permutate|pi|pop|popup|pow|powerset|powmod|prefix|print|prints|process|product|query|random|range|read|relative|remove|rename|render|repeat|replace|request|return|reverse|round|sample|saturate|script|sec|sech|select|serve|set|shl|shr|shuffle|sin|sinh|size|skewness|slice|sort|spin|split|sqrt|squeeze|stack|strip|sub|suffix|sum|switch|symbols|symlink|sys|take|tan|tanh|terminal|terminate|to|true|truncate|try|type|unclip|union|unique|unless|until|unzip|upper|values|var|variance|volume|webview|while|with|wordwrap|write|xnor|xor|zip)\b/,
+        alias: 'keyword'
+    },
+
+    'sugar': {
+        pattern: /\->|=>|\||\:\:/,
+        alias: 'important'
+    },
+
+    'symbol': {
+        pattern: /<\:|\-\:|ø|@|#|\+|\||\*|\$|\-|\%|\/|\.\.|\^|~|=|<|>|\\|\-\-\-/
```

Comment:
ESLint gives us an error here because the `\-\-\-` alternative at the end is effective ignored because of the `\-` alternative. Try highlighting the text `---`. You'll see that it gets tokenized as 3 times `-`.

To fix this, just move the `---` alternative before the `-` alternative.

Also, please only escape punctuation if necessary. Unnecessary escaping just introduces noise that make regexes a little harder to read.
```suggestion
        pattern: /<:|-:|ø|@|#|\+|\||\*|\$|---|-|%|\/|\.\.|\^|~|=|<|>|\\/
```

Reply:
I know, it looks messy. The result is like it is simply because I've used pretty-much the same syntax/patterns for different highlighters and I guess some of the syntax is a leftover of what other syntaxes expected.

Fixed :)